02-express-tutorial/routes/products.js

```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+const router = express.Router();
+
+const {
+  getProducts,
+  getProduct,
+  findProduct
+} = require("../controllers/products");
+
+// router.get("/api/v1/test", (req, res) => {
+//   res.status(200).json({ message: "It worked!" });
+// });
+
+router.get("/", getProducts);
+
+router.get("/:productID", getProduct);
+
+router.get("/query", findProduct);
```

Comment:
You need to move this route before this `router.get("/:productID", getProduct)`, because when we call the endpoint http://localhost:3000/api/v1/products/query, the express will match with the route http://localhost:3000/api/v1/products/1, because it thinks the "query" is the product id.
If you move it, it will not happen because Express will match the route "products/query" before the "products/:productId" route.